src/org/labkey/test/components/QueryMetadataEditorPage.java

```diff
@@ -0,0 +1,91 @@
+package org.labkey.test.components;
+
+import org.labkey.test.Locator;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class QueryMetadataEditorPage extends DomainDesignerPage
+{
+    public QueryMetadataEditorPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    @Override
+    public void clickFinish()
+    {
+        scrollIntoView(elementCache().finishButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().finishButton()));
+        click(Locator.button(elementCache().finishButton().getText()));
+    }
+
+    public void reset()
+    {
+        scrollIntoView(newElementCache().resetButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(newElementCache().resetButton()));
+        click(Locator.button(newElementCache().resetButton().getText()));
+        click(Locator.button("Reset")); // Reset confirmation on the confirm modal
+    }
```

Comment:
This pattern will find the same button four times and unnecessarily get its text once (`WebElement.getText` is a fairly time consuming operation and should be avoided).
 - You can eliminate the redundant element finding by refactoring the element cache away from using methods (See corresponding feedback below).
 - The `getText` is just unnecessary; you can click a `WebElement` directly.
 - `newElementCache` shouldn't need to be called directly. Doing so prevents the cache from working correctly. (See `elementCache` feedback)
```suggestion
    public void reset()
    {
        scrollIntoView(elementCache().resetButton);
        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().resetButton));
        elementCache().resetButton.click();
        click(Locator.button("Reset")); // Reset confirmation on the confirm modal
    }
```